lib/src/hex_util.rs

```diff
@@ -14,7 +14,17 @@
 
 #![allow(missing_docs)]
 
-fn to_reverse_hex_digit(b: u8) -> Option<u8> {
+pub fn to_reverse_hex_digit(b: u8) -> u8 {
+    let offset = match b {
+        b'0'..=b'9' => b - b'0',
+        b'A'..=b'F' => b - b'A' + 10,
+        b'a'..=b'f' => b - b'a' + 10,
+        b => return b,
```

Comment:
I’m guessing this can result in more efficient code (in terms of code size and runtime performance), since it means the function never panics.

Reply:
> I’m guessing this can result in more efficient code (in terms of code size and runtime performance), since it means the function never panics.

When the input is valid (the fast path) the code stays the same. Using `unreachable!()` here would prevent risking returning an invalid result in case the input is invalid.

Reply:
Right. I just mean that having to handle the potential of a panic in a function call can potentially result in worse code generation. I don’t know if it’s meaningful in this case, though.

Reply:
Yes, having a panicking path here results in unwinding codegen that propagates through callers. Its not really relevant for us here.